### PR TITLE
Merge develop into master (after a while)

### PR DESCRIPTION
**Pull request by volhovm**
Develop has not been merged into master for a year or so. Ideally master should contain all the changes from develop. 

Highlighting some changes in master (but not develop) that I found hard to reconcile with develop:
- Move tests to sub-directories
- SRS refactoring in poly-commitment/src that gets rid of srs.rs etc.
  - This was annoying to resolve since it's a big refactoring.
- posedidon changes: see commit 34dd76054a28eea4ebe7720c9b41963707cefcfa and around
- `chunked.rs` was removed from poly-commitment
- Precomputed srs stuff, master has precomputed srs for bn256 too.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -83,6 +84,11 @@ jobs:
         with:
           submodules: recursive
 
+      - name: Set up NodeJS
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node }}
```
Comment (shimkiv) [inline .github/workflows/ci.yml:90]: We need to also set the node version like here https://github.com/o1-labs/proof-systems/blob/befc0ae73fbd9bf6f16d424e46e0726a724791cb/.github/workflows/ci.yml#L81
I believe this step is needed for new runners that don't provide NodeJS runtime by default (any new custom runner).

**Review by shimkiv — CHANGES_REQUESTED** [review]
Small change requested, other than that looks good CI wise.

File: curves/src/pasta/fields/mod.rs [hunk 2]
```diff
@@ -43,3 +43,7 @@ pub trait SquareRootField: Field {
     /// Sets `self` to be the square root of `self`, if it exists.
     fn sqrt_in_place(&mut self) -> Option<&mut Self>;
 }
+
+// @volhovm: do we need this?
```
Comment (dannywillems) [inline curves/src/pasta/fields/mod.rs:47]: It seems tests are duplicated. See `curves/tests`.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
+name: proof-systems-github-runners
```
Comment (dannywillems) [inline docker-compose.yml:1]: What's this? I have never seen it before.

File: curves/src/pasta/fields/tests.rs [hunk 4]
```diff
@@ -0,0 +1,5 @@
+use crate::pasta::fields::{Fp as Fr, Fq};
```
Comment (dannywillems) [inline curves/src/pasta/fields/tests.rs:1]: Seems duplicated. See `curves/tests`.

File: poly-commitment/src/tests/mod.rs [hunk 5]
```diff
@@ -0,0 +1,3 @@
+mod batch_15_wires;
```
Comment (dannywillems) [inline poly-commitment/src/tests/mod.rs:1]: This can be removed. Tests are in `poly-commitment/tests`, not in `poly-commitment/src/tests` anymore.

**Review by dannywillems — APPROVED** [review]
Only small comments that can be done in a follow-up.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (volhovm) [reply .github/workflows/ci.yml:91]: Addressed, re-review?

File: curves/src/pasta/fields/tests.rs (hunk 4 above)
Comment (volhovm) [reply curves/src/pasta/fields/tests.rs:1]: Yep, nice catch, thanks! Duplicates removed now.

File: docker-compose.yml (hunk 3 above)
Comment (volhovm) [reply docker-compose.yml:1]: Added here: https://github.com/o1-labs/proof-systems/pull/2570/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3

(Dunno what this is, all questions to Serhii :grin:)

**Review by shimkiv — APPROVED** [review]
LGTM!

**Comment by dannywillems** [comment]
You'll need to rebase on top of master to get the new CI configuration (sorry :( )

**Comment by dannywillems** [comment]
I updated the CI branch protection. `node` was added to version 20, and therefore branch protection had to changed.
I removed:
```
Run checks and tests (1.73, 4.14) 
Run checks and tests (1.73, 4.14) 
Run checks and tests (1.73, 4.14) 
Run checks and tests (1.73, 4.14)
```
and added
```
CI / Run checks and tests (1.71, 4.14, 20) 
CI / Run checks and tests (1.72, 4.14, 20) 
CI / Run checks and tests (1.73, 4.14, 20) 
CI / Run checks and tests (1.74, 4.14, 20)  
```

**Comment by dannywillems** [comment]
Feel free to merge @volhovm

**Pull request merged**